app/javascript/actionsPanel/__tests__/actionsPanel.test.js

```diff
@@ -0,0 +1,236 @@
+      let category = 'thumbsup';
+      fetch.mockResponse(sampleResponse(category));
+      addReactionButtonListeners();
+
+      const thumbsupButton = document.querySelector(
+        `.reaction-button[data-category="${category}"]`,
+      );
+      thumbsupButton.click();
+      expect(thumbsupButton.classList).toContain('reacted');
+
+      category = 'thumbsdown';
+      const thumbsdownButton = document.querySelector(
+        `.reaction-button[data-category="${category}"]`,
+      );
+      thumbsdownButton.click();
+      expect(thumbsdownButton.classList).toContain('reacted');
+
+      category = 'vomit';
+      fetch.resetMocks();
+      fetch.mockResponse(sampleResponse(category));
+      const vomitButton = document.querySelector(
+        `.reaction-vomit-button[data-category="${category}"]`,
+      );
+      vomitButton.click();
+      expect(vomitButton.classList).toContain('reacted');
+    });
+    test('it unmarks the proper reaction(s) when positive/negative reactions are clicked', async () => {
+      let category = 'thumbsup';
+      fetch.mockResponse(sampleResponse(category));
+      addReactionButtonListeners();
+      const thumbsupButton = document.querySelector(
+        `.reaction-button[data-category="${category}"]`,
+      );
+      thumbsupButton.click();
+
+      category = 'thumbsdown';
+      fetch.resetMocks();
+      fetch.mockResponse(sampleResponse(category));
+      const thumbsdownButton = document.querySelector(
+        `.reaction-button[data-category="${category}"]`,
+      );
+      thumbsdownButton.click();
+      expect(thumbsupButton.classList).not.toContain('reacted');
+
+      fetch.resetMocks();
+      category = 'thumbsup';
+      fetch.mockResponse(sampleResponse(category, false));
+      thumbsupButton.click();
+      expect(thumbsdownButton.classList).not.toContain('reacted');
+      expect(thumbsupButton.classList).toContain('reacted');
+
+      category = 'vomit';
+      fetch.resetMocks();
+      fetch.mockResponse(sampleResponse(category));
+      const vomitButton = document.querySelector(
+        `.reaction-vomit-button[data-category="${category}"]`,
+      );
+      vomitButton.click();
+      expect(vomitButton.classList).toContain('reacted');
+      expect(thumbsupButton.classList).not.toContain('reacted');
+    });
+  });
+});
+
+describe('addAdjustTagListeners()', () => {
+  describe('when the user is tag moderator of #discuss', () => {
+    beforeEach(() => {
+      const tagName = 'discuss';
+      document.body.innerHTML = `
+      <a href="/t/${tagName}" class="tag">${tagName}</a>
+      <button class="adjustable-tag" data-adjustment-type="subtract" data-tag-name="${tagName}">
+        #${tagName}
+      </button>
+      <form id="adjustment-reason-container" class="adjustment-reason-container hidden">
+        <textarea class="crayons-textfield" placeholder="Reason for tag adjustment" id="tag-adjustment-reason" required></textarea>
```

Comment:
```suggestion
        <textarea class="crayons-textfield" placeholder="Reason for tag adjustment (required)" id="tag-adjustment-reason" required></textarea>
```

I tried submitted tag changes and wasn't aware this is was required. This minor change would help.

Reply:
Thanks!